code/controllers/subsystem/pai.dm

```diff
@@ -34,12 +36,23 @@ SUBSYSTEM_DEF(pai)
 		var/datum/pai_candidate/candidate = locate(href_list["candidate"]) in candidates
 		var/option = href_list["option"]
 		var/t = ""
+		var/list/pai_color_predetermined = list("Malware Red" = "#DA1B1B", "Data Green" = "#2DDF1D", "Holonet Blue" = "#13A5FA", "Input Yellow" = "#FCEE2E", "Storagespace Orange" = "#E9901D", "Output Purple" = "#EE1AE3")
 
 		switch(option)
 			if("name")
 				t = sanitize_name(stripped_input(usr, "Enter a name for your pAI", "pAI Name", candidate.name, MAX_NAME_LEN),allow_numbers = TRUE)
 				if(t)
 					candidate.name = t
+			if("color")
+				var/choice = tgui_input_list(usr, "Pick a colour for your pAI", "Interface Color",list("Malware Red", "Data Green", "Holonet Blue", "Input Yellow", "Storagespace Orange", "Output Purple", "Custom"))
+				if(choice == "Custom")
+					choice = input(usr, "Enter a color for your pAI.", "pAI color", candidate.color) as color|null
+					if(is_color_dark(choice, 0.5))
+						tgui_alert(usr, "The colour is too dark! Pick something lighter!", "Error", list("OK"))
+					else
+						candidate.color = choice
```

Comment:
Could you make choice into a new var here? since it's no longer a name from list and rather a raw color. Makes things less confusing if one var is only ever one class of thing

Reply:
(what)

Reply:
Ok so like, right now `choice` is the color name the choose from that list you have
It also becomes the color they choose from the color wheel thing byond has
I'd prefer those two values to be called different things. They're both choices, but they contain different types of data.

If someone's reading your code I want them to be able to think of `choice` as a name and `whatever you create for input` as a color, rather then thinking of choice as sometimes a color and sometimes a name.

It's not horribly important here, but it's a bad pattern to get into so I'ma be a hardass.